Exporting calendar redesign

NIDHI2023 opened the pull request:
### Summary

Implements Lucy's design changes for the exporting Google/Apple calendar feature.

Originals:

![image](https://github.com/user-attachments/assets/377c198c-4cfa-46e6-a863-2a582a6c0e9e)
![image](https://github.com/user-attachments/assets/5ea0241c-f9fa-4cc8-8b88-5ff6ff393cf1)

Figma redesigns:

![image](https://github.com/user-attachments/assets/91a5d036-40c3-4404-9be5-50792e392bbd)
![image](https://github.com/user-attachments/assets/0f4b52e5-a527-4335-b359-94ede1839943)
![image](https://github.com/user-attachments/assets/9a5c6a01-98e6-492e-a27e-549f074ea106)

### Actual Changes

- Got rid of bookmark
- Changed TA unfinished indicator to be a red dot when there is someone in "Ahead"

![image](https://github.com/cornell-dti/office-hours/assets/57076607/4d88b754-cf3b-491a-8af6-cd7afa47ef95)

![image](https://github.com/cornell-dti/office-hours/assets/57076607/efcfec9e-b907-4a11-9bba-1566b38ed19b)

Redesigned modal:

- Buttons are now vertical
- Functionality is the same
- Got rid of repeating office hour title, now is just course code and OH name
- Added date and time as subtitle

![image](https://github.com/cornell-dti/office-hours/assets/57076607/a0b9432e-fe3f-43f8-803c-c69065cb2c10)

With overflowing text:

![image](https://github.com/cornell-dti/office-hours/assets/57076607/b4473cb4-0e7d-4969-8681-f42367b8ac1e)


### Test Plan

- Change the "Ahead" number of students and make sure red dot appears when there is at least one, otherwise should go away
- Make sure Calendar Export modal shows expected name, date, and times for any office hour
- Check if styling remains consistent for long OH names
- Check if buttons work as expected

### Notes

- Added dependency (ran yarn add @mui/icons-material)

### Breaking Changes 

None

### Checklist

- [x] I have updated the documentation accordingly.
- [ ] My PR adds a @ts-ignore

stevenyuser reviewed (commented):
Hi Nidhi, the redesign looks pretty good! I think the design very closely matches Lucy's Figma designs and looks a lot better than the previous version.

I have some minor suggestions/bugs:
- There's a CSS error when the location of the OH session is two lines
<img width="395" alt="image" src="https://github.com/cornell-dti/office-hours/assets/77898813/d971f0fe-fc21-444a-8c56-570f652ad206">

- When clicking "Add to GCal", I think it should open a GCal link on a new tab instead of opening the GCal link on the same tab. One way to do this is by changing your current GCal Button from this:
```js
<Button href={getGoogleCalendarLink()} className="export-btn" >
    <Button.Content icon>
        <Icon name='google' />
        Add to GCal
    </Button.Content>
</Button>
```
to this:
```js
<Button
    onClick={() => window.open(getGoogleCalendarLink(), "_blank")}
    className="export-btn"
>
    <Button.Content icon>
        <Icon name="google" />
        Add to GCal
    </Button.Content>
</Button>
```

Other than that, everything looks great! Good job!

burninglilies reviewed (commented):
Hi Nidhi! Great work on this speedy PR. Aside from Stevens suggestion about opening Calendar in new tab and trying to make Session names one line I just have one more suggestion. Since we are no longer using the bookmark icon nor Google or Apple Icon it would also be a good to remove them from our media as well if they are not being used.

NIDHI2023 commented:
- Fixed Steven's suggestions for styling issues and opening a new tab

![image](https://github.com/user-attachments/assets/e6d20ca6-4b22-4ba0-ade1-159052556dae)

- Added a small fade in animation for the calendar modal only (did not touch the others in case this animation is not needed, but can be easily applied to modals in general too)

-Unneeded icons were deleted from /media

rgu0114 commented on src/components/includes/CalendarExportModal.tsx:
```diff
@@ -142,43 +149,51 @@ const CalendarExportModal = ({
-                        {isDayExport ? 
-                            <div className='Title'>{getDateString()} Office Hours</div> :
+                        <img src={CalIcon} alt='Calendar export icon' id="calIcon" />
+                        {isDayExport ?
```
nit: for code readability, make it more clear what's being conditionally rendered. For longer components like these, it might be helpful to extract them into helper functions. So in this case I might do the following instead...

```
const DayExport = () => { <div> ... </div> }

const SessionExport = () => { <div> ... </div> }

{isDayExport ? `<DayExport />` : `<SessionExport />`}


```

rgu0114 replied:
also there's quite a bit of repeated code here – you can try destructuring `currentExportSessions` earlier in the code or moving `<div className='Title'...` outside the ternary, for example. Play around with these options and see which combination looks cleanest to you!

rgu0114 commented on src/components/includes/CalendarSessionCard.tsx:
```diff
@@ -100,12 +93,16 @@ const CalendarSessionCard = (props: {
-                        alt='Export to calendar'
-                        className='CalendarExportIcon'
+
+                    {numAhead > 0 ? (<div className={'Indicator ' + props.status}>
```
similar code style thing here: use `condition && component` instead of `condition ? component : null`

rgu0114 reviewed (changes requested):
Great work here Nidhi! Agree with Steven and Lily's comments. The functionality of the redesign works as intended, and the ui matches the designs closely (very clean btw). I also liked how comprehensively you tested this + listed out the steps. Aside from a couple nit pick code style changes, could you attach screenshots of the before vs. after and the figma designs for future reference? Other than that, should be ready to merge soon!

rgu0114 reviewed (approved):
👍
